gdx/src/com/badlogic/gdx/math/Intersector.java

```diff
@@ -198,6 +198,8 @@ public static boolean intersectPolygons (Polygon p1, Polygon p2, Polygon overlap
 			floatArray.clear();
 		}
 		// Check for 3 or more vertices needed due to floating point precision errors
```

Comment:
I would leave this comment where it was before (over the line "if (floatArray2.size >= 6) {") and add a comment for this block here along the lines of "Ensure first and last point are different to protect agains floating point precision errors".